### PR TITLE
1021 - Add multiple choice answer type for Question node type

### LINKED ISSUE #1021: Add multiple choice answer type for Question node type

### PR DESCRIPTION
**Pull request by harry02031**
## Changes
This PR allows the author to select a multiple choice answer type to the question node type (in the form of checkboxes if they want the user to select more than one answer or in the form of radio buttons if they only want the user to select one answer). The author is able to type in the values of the choices, change the ordering of the choices and also upload images for each choice if they choose. The user is able to then click on the lightbox and select answer to the question in the form of checkboxes or radios depending on whatever the author selects during the authoring process.

* Authoring: the author is able to able to store all their input information in node.typeData.options.multipleChoice
Screen Shot 2021-05-26 at 10 48 35 AM

* User: the answer selected by the user is saved in quiz[0].entries as multipleChoiceId: { [multipleChoiceId]: "response" } to make it similar to how text and audio responses are stored in the depended PR 991
Screen Shot 2021-05-26 at 10 49 21 AM

## Screenshot(Video of Feature)

https://user-images.githubusercontent.com/30706624/119707676-461d1f80-be10-11eb-8502-ed1614b3a136.mov

## Issue Linkage
Closes #1021
## PR Dependency
Depends on #994
## Automated Testing
* should be able to complete with a radio answer
* should be able to complete with a checkbox answer
* should be able to complete with a checkbox answer with images
* should be able to complete with a radio answer with images

### CODE REVIEW

**Comment by Frozemint** [comment]
Is the "Show Answer to a Previous Activity First" feature supported yet? I can't seem to be able to get it working.

Also, currently it's possible to create a multiple choice question with no options. Is this intended behaviour?

**Comment by harry02031** [comment]
Hey @Frozemint, you can git pull to get the latest version that addresses the issues you mentioned. I implemented to make "Show Answer to a Previous Activity First" feature work with multiple choice, and the multiple choice question cannot remove the last option now.

**Comment by Frozemint** [comment]
Thanks, the functionalities I mentioned works now. Would it be possible to realign the multiple choice checkbox/radio buttons in a vertical column like so? This way the style is more consistent with other multiple choice elements the user might encounter from other web pages.

image
And perhaps it might be a good idea to align the multiple choice boxes in the center, since that's where the majority of the html elements are... makes the interface easier to see and use. I made a crude approximate sketch in the screenshot below.
image

In addition, the question author textfields currently have placeholders that can become out of date when the user drag them around. It might be a good idea to not specify ordinality in the placeholder and to instead just for all the placeholders to read "multiple choice option text". This way no matter how many options the user adds/delete/reorders, the placeholder still makes sense.
<p float="left">
  
   
</p>

**Comment by harry02031** [comment]
Hi @Frozemint, I made changes to the interface to take in account for your comments.

File: templates/vue/src/components/modals/NodeModal/forms/ContentForm/MultipleChoiceForm.vue [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        {
+          id: 3,
+          imageurl: '',
+          value: '',
+          title: '3rd Choice',
+        },
+      ],
+      nextChoiceRowId: 4,
+      newChoiceRowTitle: '',
+      choiceRowsRadio: [
```
Comment (Frozemint) [inline templates/vue/src/components/modals/NodeModal/forms/ContentForm/MultipleChoiceForm.vue:122]: How come the choiceRowsRadio is an array? I'm not sure if I am understanding the code properly here, but this is the variable that tracks the _single_ selected answer for a radio button right? If a radio button only allows for a single selection, why is an array needed?

**Comment by harry02031** [comment]
Updated interface styling

File: templates/vue/src/components/modals/NodeModal/forms/ContentForm/MultipleChoiceForm.vue (hunk 1 above)
Comment (harry02031) [reply templates/vue/src/components/modals/NodeModal/forms/ContentForm/MultipleChoiceForm.vue:124]: no, that is what provides the data for the list of options that the author can put in values for radio, the variable that tracks the single selected answer for a radio button is selectedRadioForm 
Screen Shot 2021-05-27 at 3 10 09 PM

**Comment by Frozemint** [comment]
Thanks for all the small adjustments and changes you've made Harry. I'm happy with what we have at this point, and I think any further issues, such as possible code refactoring, and more tweaks and fixes for minor interface issues that are more obscure could work as a future task and PR. Let's have Aidin take a last look and we are done.

**Review by Frozemint — APPROVED** [review]

**Comment by harry02031** [comment]
Closing this PR as it's superseded by #1057

**Pull request closed without merging**